spec/models/case_log_spec.rb

```diff
@@ -395,9 +395,22 @@
         expect(record_from_db["prevloc"]).to eq(nil)
       end
 
+      it "correctly resets la if la is not known" do
+        address_case_log.update!({ previous_postcode_known: "No" })
+        address_case_log.update!({ previous_la_known: "Yes", prevloc: "Scotland" })
+        record_from_db = ActiveRecord::Base.connection.execute("select prevloc from case_logs where id=#{address_case_log.id}").to_a[0]
+        expect(record_from_db["prevloc"]).to eq("S92000003")
+        expect(address_case_log.prevloc).to eq("Scotland")
+
+        address_case_log.update!({ previous_la_known: "No" })
+        record_from_db = ActiveRecord::Base.connection.execute("select prevloc from case_logs where id=#{address_case_log.id}").to_a[0]
+        expect(address_case_log.prevloc).to eq(nil)
+        expect(record_from_db["prevloc"]).to eq(nil)
+      end
+
       it "changes the prevloc if previous postcode changes from not known to known and provided" do
         address_case_log.update!({ previous_postcode_known: "No" })
-        address_case_log.update!({ prevloc: "Westminster" })
+        address_case_log.update!({ previous_la_known: "Yes", prevloc: "Westminster" })
```

Comment:
doesn't reset it if we update it like this, but not sure if that would ever be the case:
`address_case_log.update!({ previous_postcode_known: "No",  previous_la_known: "Yes", prevloc: "Westminster" })`

Reply:
Could be attempted via the API but I think in that case a validation error would be more appropriate than a reset?

Reply:
Yes (Patch/Put request)

Reply:
I think clearing the previously inferred values of LA actually is expected if the dependency changes (but looks like we got this wrong initially):

https://communities-govuk.slack.com/archives/C01QGR78H9S/p1645532079426639

Reply:
Yeah, we can clear previously inferred value, but not the one that is provided in the current request